cores/arduino/stm32/twi.c

```diff
@@ -170,53 +175,56 @@ void i2c_custom_init(i2c_t *obj, i2c_timing_e timing, uint32_t addressingMode, u
 #if defined I2C1_BASE
   // Enable I2C1 clock if not done
   if (obj->i2c == I2C1) {
-      __HAL_RCC_I2C1_CLK_ENABLE();
-      __HAL_RCC_I2C1_FORCE_RESET();
-      __HAL_RCC_I2C1_RELEASE_RESET();
-      obj->irq = I2C1_EV_IRQn;
-#ifdef STM32F1xx
-      obj->irqER = I2C1_ER_IRQn;
-#endif
-      i2c_handles[0] = handle;
+    __HAL_RCC_I2C1_CLK_ENABLE();
+    __HAL_RCC_I2C1_FORCE_RESET();
+    __HAL_RCC_I2C1_RELEASE_RESET();
+
+    obj->irq = I2C1_EV_IRQn;
+#if !defined(STM32F0xx) && !defined(STM32L0xx)
+    obj->irqER = I2C1_ER_IRQn;
```

Comment:
one new proposal : in every place where I2Cx_ER_IRQn is being used  couldn't you use 
#if defined (I2C1_ER_IRQn)
obj->irqER = I2C1_ER_IRQn;
#endif
this would be more future proof I think

Reply:
I2C1_ER_IRQn is a enum value this could not be done.